pkg/logging/kafka/create.go

```diff
@@ -124,6 +136,42 @@ func (c *CreateCommand) createInput() (*fastly.CreateKafkaInput, error) {
 		input.Placement = fastly.String(c.Placement.Value)
 	}
 
+	if c.ParseLogKeyvals.Valid {
+		input.ParseLogKeyvals = fastly.CBool(c.ParseLogKeyvals.Value)
+	}
+
+	if c.RequestMaxBytes.Valid {
+		input.RequestMaxBytes = fastly.Uint(c.RequestMaxBytes.Value)
+	}
+
+	if c.UseSASL.Valid && c.UseSASL.Value {
+		if c.AuthMethod.Value == "" || c.User.Value == "" || c.Password.Value == "" {
+			return nil, fmt.Errorf("the --auth-method, --username,  and --password flags must be present when using the --use-sasl flag")
+
+		}
+
+		if c.AuthMethod.Value != "plain" &&
+			c.AuthMethod.Value != "scram-sha-256" &&
+			c.AuthMethod.Value != "scram-sha-512" {
+
+			return nil, fmt.Errorf("invalid SASL authentication method: %s. Valid values are plain, scram-sha-256, and scram-sha-512", c.AuthMethod.Value)
+		}
+
+		input.AuthMethod = fastly.String(c.AuthMethod.Value)
+
+		if c.User.Valid {
+			input.User = fastly.String(c.User.Value)
+		}
+
+		if c.Password.Valid {
+			input.Password = fastly.String(c.Password.Value)
+		}
+	}
+
+	if !c.UseSASL.Value && (c.AuthMethod.Value != "" || c.User.Value != "" || c.Password.Value != "") {
```

Comment:
Good question. It's not necessary. The reason I added this was trying avoid unexpected results and unnecessary debugging. My reasoning was if a user gives the auth method, username, or password they probably meant to enable SASL and by accepting the *invalid* set of options we've possibly added to the confusion. The hope is by enforcing this at creation time we're saving them and possibly ourselves some time. Happy to remove it though if you think it's not worth it.

Reply:
It is probably worth giving the user feedback when the options they used make no sense.  My $0.02 it should stay.  The more Go Way to do it would be to move this invalid case as far up in the function as possible.